config/types/files.go

```diff
@@ -45,80 +56,137 @@ type Remote struct {
 	Verification Verification `yaml:"verification"`
 }
 
-type FileUser struct {
-	Id int `yaml:"id"`
+type Directory struct {
+	Filesystem string    `yaml:"filesystem"`
+	Path       string    `yaml:"path"`
+	User       FileUser  `yaml:"user"`
+	Group      FileGroup `yaml:"group"`
+	Mode       int       `yaml:"mode"`
 }
 
-type FileGroup struct {
-	Id int `yaml:"id"`
+type Link struct {
+	Filesystem string    `yaml:"filesystem"`
+	Path       string    `yaml:"path"`
+	User       FileUser  `yaml:"user"`
+	Group      FileGroup `yaml:"group"`
+	Hard       bool      `yaml:"hard"`
+	Target     string    `json:"target"`
 }
 
 func init() {
-	register2_0(func(in Config, ast validate.AstNode, out ignTypes.Config, platform string) (ignTypes.Config, report.Report, validate.AstNode) {
-		r := report.Report{}
-		files_node, _ := getNodeChildPath(ast, "storage", "files")
-		for i, file := range in.Storage.Files {
-			file_node, _ := getNodeChild(files_node, i)
-			newFile := ignTypes.File{
-				Node: ignTypes.Node{
-					Filesystem: file.Filesystem,
-					Path:       file.Path,
-					User:       ignTypes.NodeUser{ID: &file.User.Id},
-					Group:      ignTypes.NodeGroup{ID: &file.Group.Id},
+	register2_0(convertFiles)
```

Comment:
Any reason to break into a named function? I'm not against it, but maybe split into two seperate commit (break out into named fn, then the 2.1.0 changes) to make diff viewing a little easier.

Reply:
No I was trying to hunt down a problem. A later commit should undo this, I'll clean up the history